00XX-common-ledger-protocol/00XX-common-ledger-protocol.md

```diff
@@ -0,0 +1,273 @@
+
+`Message` is used for sending information to the peer. It contains no message
+data, only side protocol data. ILQP packets are attached under the protocol
+`ilp`.
+
+#### Response
+
+- `Response` is returned if the peer acknowledges the `Message`. If the peer
+  has data to send in reply (e.g. a quote response), it is carried in the side
+protocol data.
+
+- `Error` is returned if the peer is not able to process the message, or there
+  was an unexpected error and further messages should not be sent.
+
+### Prepare
+
+`Prepare` is used to create a transfer on the bilateral ledger. The message
+data contains `transferId`, `amount`, `executionCondition`, and `expiresAt`.
+`Prepare` is a request with side effects, because it creates a transfer with
+the given details. This transfer begins in the `prepared` state.
+
+`transferId` is a 128-bit unique ID that references the transfer. `amount` is
+a 64-bit integer denominated in ledger base units. The ledger base units can be
+anything, so long as both parties on the bilateral ledger agree on their
+meaning. Examples of base units are "micro-XRP," "satoshi," or "nano-dollars
+settled over paypal." `executionCondition` is a 256-bit integer containing the
+SHA-256 hash of this conditional transfer's fulfillment.  `expiresAt` is a
+ASN.1 UTC GeneralizedTime containing the expiry of this transfer.
+
+ILP payment packets are attached to the side protocol data under the protocol
+`ilp`. 
+
+#### Response
+
+- `Response` is returned if the peer acknowledges the `Prepare`. This means the
+  transfer is now `prepared` and has been applied to the balance. It may carry
+side protocol data, for example a payment channel claim.
+
+- `Error` is returned if the peer does not accept the transfer. This could be
+  because it is incompatible with an existing transfer with the same ID (see
+Idempotency below), or because it exceeds some constraint your peer has placed.
+The amount could be too high, or could exceeds the balance.
+
+#### Idempotency
+
+If a transfer with the given ID already exists in any state, a new transfer
+should not be created. If the packet (including side protocol data) matches an
+existing transfer exactly and that transfer is in the `prepared` state, a
+`Response` should be returned. If the packet (including side protocol data) shares
+the ID of an existing transfer but other details do not match, an `Error`
+should be returned. If the packet (including side protocol data) exactly
+matches an existing transfer but that transfer is in the `fulfilled` state or
+the `rejected` state, an `Error` should be returned.
+
+#### Expiry
+
+After the `expiresAt` date is reached, the transfer can no longer be fulfilled.
+If one party on the CLP connection is keeping authoritative state, they should
+send a `Reject` request to the other party. If both parties are keeping
+authoritative state, they can independently expire the transfer (set the state
+to `rejected` and roll it back) automatically.
```

Comment:
Uses same `transferId` as original `Prepare`?

Reply:
So you're proposing to make sending the 'Reject' optional, in this case? But only if you're sure the preparer is also keeping time? How can you know that? Wouldn't it be simpler to leave it as a Prepare that has had a Response (Ack) should always get either a Fulfill or a Reject?

Reply:
whether one party is authoritative is already something that has to be decided out of band. If only one party is authoritative, they don't need to wait for RPC responses from their non-authoritative peer, for instance. If the peers know they're both authoritative, it's fine for them to eschew the useless Reject messages for timeout, because they both know that it's timing out

Reply:
That does simplify things, because that removes the only ledger-level `rejectionReason`. It means you can remove https://github.com/interledger/rfcs/blob/master/asn1/CommonLedgerProtocol.asn#L89 in favor of putting the rejectionReason (ILP-error) inside the 'ilp' entry in protocolData